### PR TITLE
Update heredoc detection to add additional edge cases

### PR DESCRIPTION
**Pull request by benwells**
The following examples should all detected as heredoc

1. "=" at the end of the key

```
test=<<EOF
hello world
EOF
```
2. any number of spaces between the equals-containing key and <<

```
test=      <<EOF
hello world
EOF
```

3. case where there the delimeter itself has an equals (think base64 encoded random string)

```
MY_KEY<<sdfsdf=
hellow world
sdfsdf=
```

We also want to correctly detect "normal" single line key/value style where "<<" is in the value. 

`stdout=hi there << hello world`

### CODE REVIEW

File: src/Runner.Worker/FileCommandManager.cs [hunk 1]
```diff
@@ -383,6 +395,34 @@ public EnvFileKeyValuePairs(IExecutionContext context, string filePath)
...
+                throw new ArgumentOutOfRangeException("Whitespace check: Positions should be within the string length.");
+            }
+
+            // Ensure pos1 is always the smaller position
+            if (pos2 < pos1)
+            {
+                int temp = pos2;
+                pos2 = pos1;
+                pos1 = temp;
+            }
```
Comment (cory-miller) [inline src/Runner.Worker/FileCommandManager.cs:413]: I am not sure we should flip these. If equalsIndex is after heredocIndex, then we have a string like:

```
FOO << =EOF
abc
=EOF
```

I think we can just check `if pos2 < pos1` and throw an argument exception.

File: src/Runner.Worker/FileCommandManager.cs [hunk 2]
```diff
@@ -322,9 +322,21 @@ public EnvFileKeyValuePairs(IExecutionContext context, string filePath)
...
+                    // "key=<<EOF is also considered heredoc where the key contains an =.
+                    // Also we detect heredoc if whitespace chars exist between = and <<. (ex: "key      <<EOF")
+                    var isHeredoc = heredocIndex >= 0 &&
+                    (
+                        equalsIndex < 0 ||
+                        (
+                            heredocIndex > equalsIndex &&
+                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
+                        ) ||
+                        heredocIndex < equalsIndex
```
Comment (cory-miller) [inline src/Runner.Worker/FileCommandManager.cs:336]: To go along with my other comment about flipping indices, I think we can flip these to short circuit the case where heredocIndex is before equalsIndex (meaning equals was possibly part of the delimiter).
```suggestion
                        heredocIndex < equalsIndex ||
                        (
                            heredocIndex > equalsIndex &&
                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
                        )
```
```suggestion
                        (
                            heredocIndex > equalsIndex &&
                            OnlyContainsWhiteSpaceBetweenPositions(line, equalsIndex, heredocIndex)
                        ) ||
                        heredocIndex < equalsIndex
```

File: src/Runner.Worker/FileCommandManager.cs [hunk 3]
```diff
@@ -383,6 +395,34 @@ public EnvFileKeyValuePairs(IExecutionContext context, string filePath)
             return GetEnumerator();
         }
 
+        // accepts a string, two indexes, and returns true if only whitespace chars exist between those two indexes (non-inclusive)
+        private static bool CheckIfOnlyWhitespaceBetweenPositions(string str, int pos1, int pos2)
```
Comment (cory-miller) [inline src/Runner.Worker/FileCommandManager.cs:399]:
```suggestion
        private static bool OnlyContainsWhiteSpaceBetweenPositions(string str, int pos1, int pos2)
```

File: src/Test/L0/Worker/FileCommandTestBase.cs [hunk 4]
```diff
@@ -238,6 +238,10 @@ protected void TestHeredoc_EdgeCases()
                     "MY_KEY_4<<EOF",
                     "EOF EOF",
                     "EOF",
+                    "MY_KEY_5=abc << def",
+                    "MY_KEY_6=    <<EOF",
+                    "white space test",
+                    "EOF"
```
Comment (cory-miller) [inline src/Test/L0/Worker/FileCommandTestBase.cs:244]:
```suggestion
                    "EOF"
                    "MY_KEY_7 << =EOF="
                    "abc"
                    "=EOF="
```

File: src/Test/L0/Worker/FileCommandTestBase.cs [hunk 5]
```diff
@@ -247,6 +251,8 @@ protected void TestHeredoc_EdgeCases()
                 Assert.Equal($"hello=two", _store["MY_KEY_2"]);
                 Assert.Equal($" EOF", _store["MY_KEY_3"]);
                 Assert.Equal($"EOF EOF", _store["MY_KEY_4"]);
+                Assert.Equal($"abc << def", _store["MY_KEY_5"]);
+                Assert.Equal($"white space test", _store["MY_KEY_6="]);
```
Comment (cory-miller) [inline src/Test/L0/Worker/FileCommandTestBase.cs:255]:
```suggestion
                Assert.Equal($"white space test", _store["MY_KEY_6="]);
                Assert.Equal($"abc", _store["MY_KEY_7"]);
```
